### PR TITLE
HPCC-27275 Extend io stream classes to query statistic from io

### PR DESCRIPTION
**Pull request by shamser**
Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by shamser** [comment]
It would be useful to have access to the statistics through the io stream classes.  (Planning to use for tracking file statistics in spray/despray operations).

File: dali/ft/ftbase.ipp [hunk 1]
```diff
@@ -132,7 +132,7 @@ public:
 
     unsigned getCRC()               { return crc; }
     void setCRC(unsigned long _crc)     { crc = _crc; }
-
+    virtual unsigned __int64 getStatistic(StatisticKind kind) { return stream->getStatistic(kind); }
```
Comment (ghalliday) [inline dali/ft/ftbase.ipp:135]: style: Cleaner if this is moved next to the other virtuals.  Should also consider using override on this (and the other virtuals in this class, and elsewhere).   ( I think clang may complain if override is used on some virtuals in a class but not others...)

File: system/jlib/jfile.cpp [hunk 2]
```diff
@@ -2926,6 +2930,8 @@ class CBufferedAsyncIOStream: public CBufferedFileIOStreamBase
     virtual size32_t directRead(size32_t len, void * data) { assertex(false); return 0; }           // shouldn't get called
     virtual size32_t directWrite(size32_t len, const void * data) { assertex(false); return 0; }    // shouldn't get called
     virtual offset_t directSize() { waitAsyncWrite(); return io->size(); }
+    virtual unsigned __int64 getStatistic(StatisticKind kind) { return io->getStatistic(kind); }
+
```
Comment (ghalliday) [inline system/jlib/jfile.cpp:2934]: trivial: extra newline.  Wouldn't comment if there were no other comments.

**Review by ghalliday — COMMENTED** [review]
@shamser looks good.  A couple of minor comments.

**Review by ghalliday — APPROVED** [review]
@shamser please squash

**Pull request merged**